File: Leetcode/Python3/209. Minimum Size Subarray Sum/solution.py
# https://leetcode.com/problems/minimum-size-subarray-sum/


from typing import List


class Solution: # Sliding window
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        min_length = float("inf")
        total = left = 0
        for idx in range(len(nums)):
            total += nums[idx]
            while total >= target:
                min_length = min(min_length, idx - left + 1)
                total -= nums[left]
                left += 1
        return min_length if min_length != float("inf") else 0

File: Leetcode/C++/209. Minimum Size Subarray Sum/solution.cpp
// https://leetcode.com/problems/minimum-size-subarray-sum/

#include <algorithm>
#include <climits>
#include <vector>

using namespace std;

class Solution
{
public:
    int minSubArrayLen(int target, vector<int> &nums)
    {
        int left = 0, total = 0;
        int min_length = INT_MAX;
        for (int right = 0; right < nums.size(); right++)
        {
            total += nums[right];
            while (total >= target && left <= right)
            {
                min_length = min(min_length, right - left + 1);
                total -= nums[left];
                left += 1;
            }
        }
        return min_length == INT_MAX ? 0 : min_length;
    }
};